### PR TITLE
feat: add ocm component to the controller

### PR DESCRIPTION
**Pull request by Skarlso**
## Description

Fixes https://github.com/open-component-model/ocm-project/issues/197

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)
> Remove if not applicable

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Skarlso** [comment]
The generated component version looks okay.

```yaml
component:
  componentReferences: []
  creationTime: "2024-06-13T12:45:12Z"
  name: ocm.software/ocm-controller
  provider: ocm.software
  repositoryContexts: []
  resources:
  - access:
      localReference: sha256.6b8b5be256cc1cdfbd0411e52b1639190423bb9b319722f41b784cd7f18cdc6c
      mediaType: application/octet-stream
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: genericBlobDigest/v1
      value: 6b8b5be256cc1cdfbd0411e52b1639190423bb9b319722f41b784cd7f18cdc6c
    extraIdentity:
      architecture: amd64
      os: linux
    labels:
    - name: downloadName
      value: ocm-controller
    name: ocm-controller
    relation: local
    type: executable
    version: v0.23.3-rc.1
  - access:
      localReference: sha256.d35e45ee7bf06d9ec91cc00eca762260d3f0b6b1ea35c2f90bfb45136d1b6ed8
      mediaType: application/octet-stream
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: genericBlobDigest/v1
      value: d35e45ee7bf06d9ec91cc00eca762260d3f0b6b1ea35c2f90bfb45136d1b6ed8
    extraIdentity:
      architecture: arm64
      os: linux
    labels:
    - name: downloadName
      value: ocm-controller
    name: ocm-controller
    relation: local
    type: executable
    version: v0.23.3-rc.1
  - access:
      localReference: sha256.a1209d06c14650b5c9ac28143e545e2100bb5ff1a3d9c4596fb2796f391f0b77
      mediaType: application/octet-stream
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: genericBlobDigest/v1
      value: a1209d06c14650b5c9ac28143e545e2100bb5ff1a3d9c4596fb2796f391f0b77
    extraIdentity:
      architecture: arm64
      os: darwin
    labels:
    - name: downloadName
      value: ocm-controller
    name: ocm-controller
    relation: local
    type: executable
    version: v0.23.3-rc.1
  - access:
      localReference: sha256.71461d2c934736d6d64c18232a44e65a5485899982cc967d1813f2275b09db42
      mediaType: application/octet-stream
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: genericBlobDigest/v1
      value: 71461d2c934736d6d64c18232a44e65a5485899982cc967d1813f2275b09db42
    extraIdentity:
      architecture: amd64
      os: darwin
    labels:
    - name: downloadName
      value: ocm-controller
    name: ocm-controller
    relation: local
    type: executable
    version: v0.23.3-rc.1
  - access:
      localReference: sha256.089d291d6a95526fdb088dfdd180781066486ae2432e82ca441070431e9fc97a
      mediaType: application/vnd.oci.image.index.v1+tar+gzip
      referenceName: ocm.software/ocm-controller/ocm-controller-image:v0.23.3-rc.1
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: ociArtifactDigest/v1
      value: ebbd08eefe49223528d3eb1aa8ca84644e2341ac6f2f49a0cdceb665e50d73ea
    name: ocm-controller-image
    relation: local
    type: ociImage
    version: v0.23.3-rc.1
  - access:
      localReference: sha256.d0dfb157990d3517e23b55223fa458c71c9ad2675340b7bb733d6126360d61fc
      mediaType: application/vnd.oci.image.manifest.v1+tar+gzip
      referenceName: ocm.software/ocm-controller/ocm-controller:v0.23.3-rc.1
      type: localBlob
    digest:
      hashAlgorithm: SHA-256
      normalisationAlgorithm: ociArtifactDigest/v1
      value: dcf4dc33399796fd4f03aa894e58b71657465211f1e4e68c8d95ebeb06840b48
    name: charts
    relation: local
    type: helmChart
    version: v0.23.3-rc.1
  sources:
  - access:
      commit: 20ead4b48c11e2c586e283bccd8dd56859aa9e6d
      repoUrl: github.com/open-component-model/ocm-controller
      type: github
    name: source
    type: filesystem
    version: v0.23.3-rc.1
  version: v0.23.3-rc.1
meta:
  schemaVersion: v2

```

File: component/Makefile [hunk 1]
```diff
@@ -0,0 +1,149 @@
+NAME      = ocm-controller
```
Comment (hilmarf) [inline component/Makefile:1]: here it is `ocm-controller`

File: component/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# ocm - command line interface
+
+This folder contains an example how to build a component version for the OCM cli.
```
Comment (hilmarf) [inline component/README.md:3]: but here it's `OCM cli`

either or... controller or ocm-cli?

**Review by morri-son — APPROVED** [review]
lgtm

File: component/README.md (hunk 2 above)
Comment (Skarlso) [reply component/README.md:3]: Ah, Hilmar, you're the man. Gerald didn't notice. :D I'll fix it in a sec. :)

Comment (Skarlso) [reply component/README.md:3]: https://github.com/open-component-model/ocm-controller/pull/467

**Comment by Skarlso** [comment]
FFS. I forgot to add the part where I install OCM... 🤦

**Pull request merged**